toolkit/tools/grapher/grapher_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package main
+
+import (
+	"os"
+	"testing"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/pkggraph"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/pkgjson"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	logger.InitStderrLog()
+	logger.Log.SetLevel(logrus.FatalLevel)
+	os.Exit(m.Run())
+}
+
+// Make a "local" package of the given name. Each package with the same pkgName will share the same
+// requires, srpmpath, etc. The subPkgName is used to differentiate multiple run-nodes for each
+// package. The buildRequires should be common across all packages with the same pkgName.
+func makePackage(pkgName, subPkgName string, buildRequires []*pkgjson.PackageVer) *pkgjson.Package {
+	// Create a package with the given name and version
+	pkg := &pkgjson.Package{
+		Provides: &pkgjson.PackageVer{
+			Name:    subPkgName,
+			Version: "1.0",
+		},
+		SrpmPath:     pkgName + ".src.rpm",
+		RpmPath:      pkgName + ".rpm",
+		SourceDir:    pkgName + "-src",
+		SpecPath:     pkgName + ".spec",
+		Architecture: "x86_64",
+		IsToolchain:  false,
+		RunTests:     true,
+	}
+
+	pkg.BuildRequires = buildRequires
+
+	return pkg
+}
+
+// A very simiple test case to validate that the test framework is working
+func TestValidate(t *testing.T) {
+
+	testRepo := &pkgjson.PackageRepo{
+		Repo: []*pkgjson.Package{
+			makePackage("pkg1", "pkg1-a", nil),
+			makePackage("pkg1", "pkg1-b", nil),
+		},
+	}
+	depGraph := pkggraph.NewPkgGraph()
+	err := populateGraph(depGraph, testRepo)
+	require.NoError(t, err)
+
+	// Validate that the graph is not empty
+	require.Equal(t, 2, len(depGraph.AllRunNodes()))
+	require.Equal(t, 2, len(depGraph.AllBuildNodes()))
+}
+
+// Load a specially crafted repo containing several sub-packages of a single SRPM, and two malicious
+// packages that try to pollute the available remote dependencies. The test will check if the various build
+// nodes of the sub-package SRPM are all using the same set of dependencies. The other packages will add
+// new remote nodes to the graph since they may not be satisfied by the already existing nodes. Once
+// this happens, the sub-package build nodes may start using the "better" dependencys, which is not what we want.
+func TestScenarioMultiRemoteProvides(t *testing.T) {
+	anyVer := []*pkgjson.PackageVer{
```

Comment:
Could you add some info, that these are versions of build-time dependencies. For instance, call this `anyVerBR` or instead of "dep1" use "BuildRequires1". Something along these lines, just to make it clearer what we're looking at.